### PR TITLE
Create BudgetingList Component

### PR DESCRIPTION
**Pull request by Afrokk**

### CODE REVIEW

File: src/components/BudgetingList/BudgetingList.sass [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    .budgeting-list
+        li
+            padding-top: 10px
+            transition: 0.5s
+            opacity: 100
+        
+        li:first-child
+            padding-top: 0px
+    
+    .new-field
```
Comment (UsmanAJabbar) [inline src/components/BudgetingList/BudgetingList.sass:18]: .new-field? You can name this better

File: src/components/BudgetingList/BudgetingList.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            fieldType="AMOUNT"
+            formatting={true}
+            required={true}
+          />
+        </li>
+        {Fields.map((field, index) => {
+          return (
+            <li key={index} className="new-field">
+              <InputField label={field} fieldType="TEXT" />
+              <svg
```
Comment (UsmanAJabbar) [inline src/components/BudgetingList/BudgetingList.tsx:92]: I prefer you don't add SVGs directly in here

File: src/components/InputField/InputField.tsx [hunk 3]
```diff
@@ -39,6 +44,23 @@ const InputField = ({
     (required && VALIDATOR[fieldType].test(value)) ||
     (!required && !!value && VALIDATOR[fieldType].test(value));
 
+  const formatInput = (): string => {
+    if (fieldType === "AMOUNT" && !value.includes("$") && value !== "") {
+      console.log(value);
```
Comment (UsmanAJabbar) [inline src/components/InputField/InputField.tsx]: console.logs shouldn't be here...

File: src/components/InputField/InputField.tsx [hunk 4]
```diff
@@ -57,12 +84,13 @@ InputField.defaultProps = {
...
   fieldType: "TEXT",
   required: false,
+  formatting: false,
 };
 
 const VALIDATOR: { [key: string]: RegExp } = {
   POSTCODE: /^\d{5}$/,
-  AMOUNT: /^\d{1,7}$/,
-  PERCENTAGE: /^\d{1,3}%$/,
+  AMOUNT: /^\$?\d{1,7}$/,
```
Comment (UsmanAJabbar) [inline src/components/InputField/InputField.tsx:89]: I'd avoid adding a $ checker in here, instead opt to make the dollar sign or currency symbol a default on all AMOUNT fields.

File: src/components/BudgetingList/BudgetingList.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                onClick={() => removeField(index)}
+                xmlns="http://www.w3.org/2000/svg"
+                fill="none"
+                viewBox="0 0 24 24"
+                strokeWidth={1.5}
+                stroke="rgb(214, 214, 214)"
+                className="w-6 h-6"
```
Comment (UsmanAJabbar) [inline src/components/BudgetingList/BudgetingList.tsx:99]: Unused classes in here

File: src/components/InputField/InputField.tsx [hunk 6]
```diff
@@ -5,12 +5,14 @@ type InputFieldProps = {
...
   fieldType: "POSTCODE" | "AMOUNT" | "PERCENTAGE" | "TEXT";
   required: boolean;
+  formatting: boolean;
 };
 
 const InputField = ({
   label,
   fieldType,
   required,
+  formatting,
```
Comment (UsmanAJabbar) [inline src/components/InputField/InputField.tsx:15]: This should not be a prop, and instead just used the usual $ sign on AMOUNT fieldtypes.

**Review by UsmanAJabbar — CHANGES_REQUESTED** [review]
Let me know when you're done with these changes.

File: src/components/InputField/InputField.tsx (hunk 4 above)
Comment (Afrokk) [reply src/components/InputField/InputField.tsx:89]: Fixed.

**Comment by Afrokk** [comment]
> Let me know when you're done with these changes.

Done

**Comment by Afrokk** [comment]
**To Do:**
- Fix styles (overlap on custom fields)
- Add "+" button in add a field button

File: src/components/BudgetingList/BudgetingList.tsx [hunk 7]
```diff
@@ -1,30 +1,95 @@
...
+
+    /* If ID is not null, set the current field's ID to 
+    the selected element's ID */
+
+    /* This is used to tell which InputField the input is coming from.
+    The corresponding field in the budgetingData is then updated. */
+
+    //This is ignored if ID is not provided.
+    if (id !== undefined) {
+      setcurrField(id);
```
Comment (UsmanAJabbar) [inline src/components/BudgetingList/BudgetingList.tsx:59]: I prefer this to be renamed to `setCurrentField` instead

File: src/components/BudgetingList/BudgetingList.tsx [hunk 8]
```diff
@@ -33,55 +98,98 @@ const BudgetingList = (): JSX.Element => {
...
     setFields(data);
+    budgetingData.splice(index + 6, 1);
+    setBudgetingData(budgetingData);
+    setValue("");
   };
 
   return (
     <div className="budgeting-list-component">
       <ul className="budgeting-list">
         <li>
```
Comment (UsmanAJabbar) [inline src/components/BudgetingList/BudgetingList.tsx:109]: I prefer this to be generated with `Array.map`. Try using a list of objects with the key being the label and the fieldType being the type and name the variable definiting these fileds `initialData` or something

File: src/components/InputField/InputField.tsx [hunk 9]
```diff
@@ -5,24 +5,44 @@ type InputFieldProps = {
   label: string;
   fieldType: "POSTCODE" | "AMOUNT" | "PERCENTAGE" | "TEXT";
   required: boolean;
+  id: string;
+  data: (value: string, id?: string) => void 
+  className: string;
```
Comment (UsmanAJabbar) [inline src/components/InputField/InputField.tsx:10]: If you're going to pass in a className, it'd rather be a list of strings and then consumed in the React component via join, joined into a string with a space in between them.

File: src/pages/Litmus/Litmus.tsx [hunk 10]
```diff
@@ -6,27 +6,30 @@ import "./Litmus.sass";
...
+          label="Single/Married"
           options={["Single", "Married"]}
           required={true}
         />
       </div>
 
       <div className="secondary-details-container">
-        <div className="half-width"></div>
-        <div className="half-width">
+        <div id="results">
```
Comment (UsmanAJabbar) [inline src/pages/Litmus/Litmus.tsx:25]: Can't approve this for now, this has to be updated to reflect the requirements. I'll look into the actual behaviour of all of this new code during the demo.

File: src/styles/_variables.sass [hunk 11]
```diff
@@ -4,5 +4,6 @@ $primary-font: 'Montserrat', sans-serif
 // Colors
 $primary-color: #00c2ff
 $placeholder-color: #656565
+$background-color: #010101
```
Comment (UsmanAJabbar) [inline src/styles/_variables.sass:7]: I have no context of what background color this is, try making it a bit clearer bit don't vaguely name it `$background-color`

**Review by UsmanAJabbar — APPROVED** [review]
LGTM

**Pull request merged**